cmd/consensus.cpp

```diff
@@ -215,32 +215,42 @@ static void check_rlp_err(rlp::DecodingError err) {
 }
 
 // https://ethereum-tests.readthedocs.io/en/latest/test_types/blockchain_tests.html#pre-prestate-section
-void init_pre_state(const nlohmann::json& pre, IntraBlockState& state) {
+void init_pre_state(const nlohmann::json& pre, StateBuffer& state) {
     for (const auto& entry : pre.items()) {
         evmc::address address{to_address(*from_hex(entry.key()))};
-        const nlohmann::json& account{entry.value()};
-        Bytes balance_str{*from_hex(account["balance"].get<std::string>())};
+        const nlohmann::json& j{entry.value()};
+
+        Account account;
+        Bytes balance_str{*from_hex(j["balance"].get<std::string>())};
```

Comment:
Question ... from_hex returns std::optional<Bytes> ... are we sure of the sanity of the Json payload ?

Reply:
Right, I've replaced it with checked access to be on the safe side.